File: DatalakeClient/src/@types/enums/TagType.ts
export enum TagType {
	String = 0,
	Number = 1,
	Boolean = 2,
}

export function TagTypeDescription(type: TagType) {
	switch (type) {
		case TagType.String: return 'строка'
		case TagType.Number: return 'число'
		case TagType.Boolean: return 'дискрет'
		default: return '?'
	}
}

File: DatalakeApiClasses/Enums/TagType.cs
using System.Text.Json.Serialization;

namespace DatalakeApiClasses.Enums;

/// <summary>
/// Тип данных
/// </summary>
[JsonConverter(typeof(JsonStringEnumConverter))]
public enum TagType
{
	/// <summary>
	/// Строка
	/// </summary>
	String = 0,

	/// <summary>
	/// Число
	/// </summary>
	Number = 1,

	/// <summary>
	/// Логическое значение
	/// </summary>
	Boolean = 2,
}
